var/spack/repos/builtin/packages/xcb-util-xrm/package.py

```diff
@@ -0,0 +1,48 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class XcbUtilXrm(AutotoolsPackage):
+    """XCB util-xrm module provides the 'xrm' library, i.e.  utility functions
+    for the X resource manager."""
+
+    homepage = "http://www.example.com"
+    url      = "https://github.com/Airblader/xcb-util-xrm/archive/v1.2.tar.gz"
+
+    # This GitHub project includes some git submodules, which must be fetched
+    # in order to build it.
+    version('1.2',
+            git='https://github.com/Airblader/xcb-util-xrm',
+            tag='v1.2',
+            submodules=True)
```

Comment:
Any reason to use the git tag instead of the archive tarball? It's easier to checksum a tarball, which is nice for security reasons. The tarball often doesn't require any of the Autotools dependencies either.

Reply:
Thanks for your patience, I addressed your points but the last one. Specifically, according to [this conversation](https://github.com/Airblader/xcb-util-xrm/issues/77), `xcb-util-xrm` project has git submodules which are not included into the release archive. My understanding is that Spack `version` argument `submodules` is only considered when the sources are retrieved through a git repo.